File: 7-kyu/alphabetWar.js
/* Write a function that accepts fight string consists of only small letters and return who wins the fight. When the left side wins return Left side wins!, when the right side wins return Right side wins!, in other case return Let's fight again!.

The left side letters and their power:

 w - 4
 p - 3
 b - 2
 s - 1
The right side letters and their power:

 m - 4
 q - 3
 d - 2
 z - 1
The other letters don't have power and are only victims.

Example
alphabetWar("z");        //=> Right side wins!
alphabetWar("zdqmwpbs"); //=> Let's fight again!
alphabetWar("zzzzs");    //=> Right side wins!
alphabetWar("wwwwwwz");  //=> Left side wins! */

//Solution
function alphabetWar(fight) {
  let leftSidePower = {
    w: 4,
    p: 3,
    b: 2,
    s: 1,
  };

  let rightSidePower = {
    m: 4,
    q: 3,
    d: 2,
    z: 1,
  };
  let left = 0;
  let right = 0;
  fight.split("").forEach((char) => {
    if (char in leftSidePower) {
      left += leftSidePower[char];
    } else if (char in rightSidePower) {
      right += rightSidePower[char];
    }
  });

  if (left == right) {
    return "Let's fight again!";
  }

  if (left > right) {
    return "Left side wins!";
  }

  return "Right side wins!";
}

File: c/alphabet_war.c
char *alphabet_war(const char *fight)
{
    int left = 0, right = 0;
    while (*fight)
    {
        switch (*fight++)
        {
        case 'w':
            left += 4;
            break;
        case 'p':
            left += 3;
            break;
        case 'b':
            left += 2;
            break;
        case 's':
            left += 1;
            break;
        case 'm':
            right += 4;
            break;
        case 'q':
            right += 3;
            break;
        case 'd':
            right += 2;
            break;
        case 'z':
            right += 1;
            break;
        }
    }

    if (left > right)
    {
        return "Left side wins!";
    }
    else if (right > left)
    {
        return "Right side wins!";
    }

    return "Let's fight again!";
}